gnocchi/cli/metricd.py

```diff
@@ -85,6 +86,8 @@ def run(self):
             with utils.StopWatch() as timer:
                 try:
                     self._run_job()
+                except ConnectionError:
```

Comment:
Ok I see another problem here, is that you are violating separation of concern.
The metricd tool does not need to know about Redis, and anything else can be used. That means the exception must be managed in the redis driver itself, not here.

Reply:
underlying incoming/redis does not handle any exceptions, but it probably should. There has been a change in redis-py around 3.0, which hinted/suggested to catch and handle exceptions in applications and not in redis-py. I can not find the issue/pr in redis-py at the moment.

Where would you want the exception handled then?

Reply:
In such files https://github.com/gnocchixyz/gnocchi/blob/master/gnocchi/incoming/redis.py and its method.
I'm not sure which part of Redis raises `ConnectionError`, if it's all of the, you might need to decorate with a tenacity decorator to retry a few times the operations, and bail out if it fails too much.